src/main/haskell/kore/src/Kore/Step/Step.hs

```diff
@@ -139,7 +139,11 @@ transitionRule tools substitutionSimplifier simplifier axiomIdToSimplifier =
                 config
                 a
         case result of
-            Left _ -> pure []
+            Left _ -> error $
```

Comment:
Might be a good idea to also print the rule and configuration for debug purposes. Other than that, failing earlier is always better.